pkgs/development/libraries/jemalloc/default.nix

```
@@ -26,7 +26,11 @@ stdenv.mkDerivation rec {
       url = "https://github.com/jemalloc/jemalloc/commit/d00ecee6a8dfa90afcb1bbc0858985c17bef6559.patch";
       hash = "sha256-N5i4IxGJ4SSAgFiq5oGRnrNeegdk2flw9Sh2mP0yl4c=";
     })
-  ];
+    # Fix linking against LLVM's libc++
+  ] ++ lib.optional (stdenv.cc.bintools.isLLVM) (fetchpatch {
```

i think the patch can be applied unconditionally, and needs a comment to remove on next update as well as a link to `https://github.com/jemalloc/jemalloc/pull/2348`. tho, the condition will allow the change to go to master rather than staging.

and as noted previously, the condition is not correct, it needs to reference `libcxx` somehow, as shown below.

```suggestion
  ] ++ lib.optional (stdenv.cc.libcxx.isLLVM or false) (fetchpatch {
```

as that will work correctly for `libcxxStdenv`

---

Alright, moved to staging and I removed the condition. I also added a note with a link to the PR.